source/core/Variants.cpp

```diff
@@ -0,0 +1,191 @@
+};
+
+// Convert variant to data of given type. Error if the data types don't match
+VIREO_FUNCTION_SIGNATUREV(VariantToData, VariantToDataParamBlock)
+{
+    ErrorCluster *errPtr = _ParamPointer(ErrorClust);
+    if (!errPtr || !errPtr->status) {
+        TypeRef variantType = _Param(VariantType);
+        TypeRef destType = _ParamImmediate(DestData._paramType);
+
+        if (variantType->IsA(destType)) {
+            void* destData = _ParamImmediate(DestData)._pData;
+            variantType->CopyData(variantType->Begin(kPARead), destData);
+        } else {
+            errPtr->SetError(true, 91, "Variant To Data");
+        }
+    }
+    return _NextInstruction();
+}
+
+struct SetVariantAttributesParamBlock : public VarArgInstruction
+{
+    _ParamDef(TypeRef, InputVariant);
+    _ParamDef(StringRef, Name);
+    _ParamImmediateDef(StaticTypeAndData, Value);
+    _ParamDef(Boolean, Replaced);
+    _ParamDef(ErrorCluster, ErrorClust);
+    NEXT_INSTRUCTION_METHODV()
+};
+
+VIREO_FUNCTION_SIGNATUREV(SetVariantAttributes, SetVariantAttributesParamBlock)
+{
+    ErrorCluster *errPtr = _ParamPointer(ErrorClust);
+    if (!errPtr || !errPtr->status) {
+        TypeRef inputVariant = _Param(InputVariant);
+        StringRef name = _Param(Name);
+        StaticTypeAndDataPointer value = &_ParamImmediate(Value);
+        VariantAttributeManager::VariantToAttributeMapType &variantToAttributeMap = VariantAttributeManager::Instance().GetVariantToAttributeMap();
+
+        auto variantToAttributeMapIter = variantToAttributeMap.find(inputVariant);
+        if (variantToAttributeMapIter != variantToAttributeMap.end()) {
+            VariantAttributeManager::AttributeMapType *attributeMap = variantToAttributeMapIter->second;
+
+            auto pairIterBool = attributeMap->insert(VariantAttributeManager::AttributeMapType::value_type(name, value));
+            _Param(Replaced) = !pairIterBool.second;
+            if (!pairIterBool.second) {
+                pairIterBool.first->second = value;
+            }
+        } else {
+            VariantAttributeManager::AttributeMapType *attributeMap = new VariantAttributeManager::AttributeMapType;
+            (*attributeMap)[name] = value;
+            variantToAttributeMap[inputVariant] = attributeMap;
+            _Param(Replaced) = false;
+        }
+    }
+    return _NextInstruction();
+}
+
+struct GetVariantAttributesParamBlock : public VarArgInstruction
+{
+    _ParamDef(TypeRef, InputVariant);
+    _ParamDef(StringRef, Name);
+    _ParamImmediateDef(StaticTypeAndData, Value);
+    _ParamDef(Boolean, Found);
+    _ParamDef(ErrorCluster, ErrorClust);
+    NEXT_INSTRUCTION_METHODV()
+};
+
+VIREO_FUNCTION_SIGNATUREV(GetVariantAttributes, GetVariantAttributesParamBlock)
+{
+    ErrorCluster *errPtr = _ParamPointer(ErrorClust);
+    _Param(Found) = false;
+    if (!errPtr || !errPtr->status) {
+        TypeRef inputVariant = _Param(InputVariant);
+        StringRef name = _Param(Name);
```

Comment:
Name here should be optional, and support returning an array of all the attributes like the prim does.
See my other comment about having two different entry points to suit this purpose.

Reply:
Fixed in new PR. Thanks.